3 - PB/Documentazione interna/Verbali/24-02-26/24-02-26.typ

```diff
@@ -0,0 +1,62 @@
+#import "/template.typ": *
+
+#show: project.with(
+  date: "26/02/24",
+  subTitle: "Meeting di Design Thinking",
+  docType: "verbale",
+  authors: (
+    "Banzato Alessio",
+  ),
+  timeStart: "15:10",
+  timeEnd: "16:50",
+);
+
+= Ordine del giorno
+
+- Progettazione:
+  - Design pattern;
+  - Architettura;
+  - Diagramma ER.
+- To do.
+
+== Progettazione
+=== Design pattern
+
+Il gruppo ha cercato di determinare i design pattern da utilizzare nella progettazione del prodotto, e ha individuato le seguenti ipotesi:
+
+- model-view pattern, in particolare il model-view-controller (MVC);
+- repository pattern.
+
+Il gruppo individua, inoltre, la possibilità di inserire il proxy pattern nel contesto della comunicazione con le API.
+
+Sono sorti dei dubbi riguardo la possibilità che il model-view-controller sia già implementato nativamente nel framework Next.js. In questo caso la sua implementazione risulterebbe un lavoro che aggiungerebbe complessità non necessaria al prodotto. Il gruppo quindi si interroga su come comportarsi nel caso alcuni pattern siano già implementati all'interno di librerie o framework utilizzati.
+
+Durante la discussione riguardo i pattern il gruppo considera la possibilità di non definire classi all'interno del codice. Questo nasce da due osservazioni:
+
+- JavaScript non è un vero e proprio linguaggio ad oggetti;
+- gli oggetti presenti nel prodotto, intesi come elementi grafici, non hanno comportamenti particolari che necessitano di essere raggruppati e organizzati.
+
+In conseguenza di ciò, sorge la questione di come affrontare i design pattern in mancanza di vere e proprie classi, la quale verrà esposta tramite mail al #cardin.
+
+=== Architettura
+
+Il gruppo discute sulla scelta dell'architettura più adatta al prodotto da sviluppare, individuando infine l'architettura layered come opzione principale, in quanto l'architettura esagonale, alternativa presa in considerazione, pone molta attenzione sulla business logic, la quale non è presente nel capitolato in quanto le API che verificano la possibilità di compiere determinate azioni sono simulate. Inoltre, la comunicazione tra componenti è unidirezionale: da database verso la presentation logic.
```

Comment:
il fatto che non abbiamo buisness logic viene più dal fatto che per natura stessa del progetto il nostro prodotto è un visualizzatore. Quindi non abbiamo manipolazione di dati o altro.
Riformulerei la frase da business logic in poi